oxidation/libparsec/crates/serialization_format/src/protocol.rs

```diff
@@ -0,0 +1,739 @@
+                                                    name: variant.name.to_owned(),
+                                                    discriminant_value: variant.discriminant_value.to_owned(),
+                                                    fields: variant.fields.as_ref().map(|fields| {
+                                                        fields.iter()
+                                                        .filter_map(&convert_field)
+                                                        .collect()
+                                                    }),
+                                                }
+                                            }
+                                        ).collect()
+                                    }
+                                },
+                                (Some(fields), None, None) => {
+                                    GenCmdNestedType::Struct {
+                                        name: nested_type.name.to_owned(),
+                                        fields: fields
+                                            .iter()
+                                            .filter_map(&convert_field)
+                                            .collect(),
+                                    }
+                                }
+                                _ => {
+                                    panic!("{:?}::{:?}: Enum should have `variants`&`discriminant_field`, Struct should have `fields`", &cmd.req.cmd, &nested_type.name);
+                                },
+                            }
+                        }).collect()
+                    } else {
+                        vec![]
+                    }
+                };
+
+                // Told you we were going to populate `allowed_extra_types` !
+                {
+                    let mut allowed_extra_types = allowed_extra_types.borrow_mut();
+                    for nested_type in gen_nested_types.iter() {
+                        let name = match nested_type {
+                            GenCmdNestedType::Enum { name, .. } => name,
+                            GenCmdNestedType::Struct { name, .. } => name,
+                        };
+                        allowed_extra_types.insert(name.to_owned(), name.to_owned());
+                    }
+                }
+
+                let gen_req = GenCmdReq {
+                    _cmd: cmd.req.cmd.to_owned(),
+                    kind: match (&cmd.req.unit, &cmd.req.fields) {
+                        (Some(nested_type_name), None) => GenCmdReqKind::Unit {
+                            nested_type_name: nested_type_name.to_owned(),
+                        },
+                        (None, Some(fields)) => GenCmdReqKind::Composed {
+                            fields: fields.iter().filter_map(&convert_field).collect(),
+                        },
+                        (None, None) => GenCmdReqKind::Composed { fields: vec![] },
```

Comment:
This type of struct are called unit struct 

```rust
// A unit struct called `Foo`
struct Foo;
```

Reply:
There is no need to separate this type with the one with fields (otherwise we will need to handle the case of the json containing an empty `fields` field)

However I choose `Composed` name without too much thinking (I just wanted a "not unit" name), do you think it would be better named `Struct` ? or something else ?